recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -110,6 +165,25 @@ add_task(withMockPreferences(function* (mockPreferences) {
   stop.restore();
 }));
 
+// startObserver should observe changes to the default preference value.
+add_task(withMockPreferences(function* (mockPreferences) {
+  const stop = sinon.stub(PreferenceExperiments, "stop");
+  mockPreferences.set("fake.preference", "startvalue", "default");
+
+  PreferenceExperiments.startObserver("test", "fake.preference", "experimentvalue");
+
+  // Setting it to the experimental value should not trigger the call.
+  DefaultPreferences.set("fake.preference", "experimentvalue");
+  ok(!stop.called, "Changing to the experimental pref value did not trigger the observer");
+
+  // Setting it to something different should trigger the call.
```

Comment:
Testing after setting it back doesn't really matter, because by that point `stop` has already been called, which (when not mocked, like in this test) would remove the observer.

`startObserver` doesn't actually modify the preference, it only sets up the preference watcher, so the preference isn't set to the experimental value until the first `DefaultPreferences.set` call. #645 isn't finished yet, but it will need to call this function on startup for the active experiments, and I figured it'd be easier to reason about how this behaves without setting the preference than it was to figure out if setting the experiment preferences on startup was an issue.

I'll add a comment clarifying that `startObserver` does not modify the preference to these tests.